### ISSUE TITLE
Broadcasting sum

### ISSUE THREAD

**Issue by telmou** [clab/dynet#378]
Is there a way of adding (column-wise) a column vector to a matrix?
I grouped a series of possible actions my system can take as columns in the E matrix. W is the weight matrix, b is the bias vector, and v is a row vector.
```
repr = dy.tanh(W*E+b)
dy.pickneglogsoftmax(v*repr, true_answer)
```

However I get an error trying to add the b vector to the W*E matrix. Is there a way to broadcast it?This is a single example from the training dataset, so I'd prefer to avoid minibatching functions until I am actually doing minibatching.

**Comment by pmichel31415** [comment]
This is done automatically if E is shaped as a batch of vector so something like 

    E.reshape((row_dim,), batch_size=col_dim)

Then repr would have shape `((len(b),), col_dim)` (the last dim being the batch dimension).

**Comment by neubig** [comment]
Please use the `colwise_sum(W*E, b)`. Reshape is not necessary.

**Comment by telmou** [comment]
Thank you both! After some experiments, I found that @pmichel31415's solution is slightly faster. Btw, @neubig, you meant colwise_add, right?

**Comment by neubig** [comment]
@telmou Yes, I did mean that. And actually, sorry I completely forgot about this, but actually `affine_transform(b, W, E)` should also work. This might be fastest.

**Comment by telmou** [comment]
@neubig Thanks, I will try that 👍
